### PR TITLE
BT controller nordic: radio_nrf5_dppi: Use HAL for registers with sideeffects 

### PR DESCRIPTION
**Pull request by aescolar**
Use the nrfx in the controller nrf53 dppi hal, so this code can run in simulation properly, let's use the nrfx HAL functions for registers which have sideeffects (the DPPI subscribe, task and CHG registers).

These are all changes which have been done:

    Bluetooth controller nrf: Use HAL for DPPI configuration
    
    The following changes have been done:
    1.
    HAL_SW_SWITCH_GROUP_TASK_DISABLE_PPI_REGISTER_TASK(index)
      = HAL_SW_SWITCH_GROUP_TASK_DISABLE_PPI_TASK(channel)
    has been converted into
    HAL_SW_SWITCH_GROUP_TASK_DISABLE_PPI_REGISTER_TASK(index, channel)
    
    2.
    HAL_SW_SWITCH_GROUP_TASK_ENABLE_PPI_REGISTER_TASK(index)
      = HAL_SW_SWITCH_GROUP_TASK_ENABLE_PPI_TASK
    Which was only used in one place, has been replaced with the
    equivalent HAL call
    
    3.
    HAL_SW_SWITCH_GROUP_TASK_ENABLE_PPI_REGISTER_TASK(index) = 0
    was replaced with the HAL subscribe clear function
    nrf_dppi_subscribe_clear(NRF_DPPIC,
       HAL_SW_DPPI_TASK_EN_FROM_IDX(SW_SWITCH_TIMER_TASK_GROUP(index)));
    
    4.
    NRF_DPPIC->SUBSCRIBE_CHG[group].EN/DIS = 0;
    have been replaced with the equivalent HAL clear call
    
    5.
    NRF_DPPIC->TASKS_CHG[group].DIS = 1;
    have been replaced with the equivalent hal task_trigger() call
    
    6.
    Manually setting the CHG registers, has been replaced with a
    clear + add (because the hal does not have a set function yet)
    So,
      NRF_DPPIC->CHG[group] = value
    has been replaced with
      nrf_dppi_group_clear(NRF_DPPIC, group);
      nrf_dppi_channels_include_in_group(NRF_DPPIC, value, group);
    (A set function has been requested from the HAL team)
    
    Note: There is other direct registes writes to the dppi subscribe and CHG
    registers in the coded phy parts which have not been changed yet.
    
-----

    Bluetooth controller nrf: HAL for DPPI configuration cleanup
    
    Remove a few macros which are not used anywhere in the tree.
    For the sake of simplifying the header and easing its understanding.

### CODE REVIEW

File: subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/radio/radio_nrf5_dppi.h [hunk 1]
```diff
@@ -317,6 +317,20 @@ static inline void hal_sw_switch_timer_clear_ppi_config(void)
 #define SW_SWITCH_TIMER_S2_EVTS_COMP(index) \
 	(SW_SWITCH_TIMER_EVTS_COMP_S2_BASE + (index))
 
+/*
+ * Convert a dppi channel group number into the *enable* task enumerate value
+ * the nrfx hal accepts
+ */
+#define HAL_SW_DPPI_TASK_EN_FROM_IDX(index) \
+	(NRF_DPPI_TASK_CHG0_EN + index*(NRF_DPPI_TASK_CHG1_EN - NRF_DPPI_TASK_CHG0_EN))
```
Comment (cvinayak) [inline subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/radio/radio_nrf5_dppi.h:325]:
```suggestion
	(NRF_DPPI_TASK_CHG0_EN + ((index) * (NRF_DPPI_TASK_CHG1_EN - NRF_DPPI_TASK_CHG0_EN)))
```

File: subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/radio/radio_nrf5_dppi.h [hunk 2]
```diff
@@ -317,6 +317,20 @@ static inline void hal_sw_switch_timer_clear_ppi_config(void)
...
+
+/*
+ * Convert a dppi channel group number into the *disable* task enumerate value
+ * the nrfx hal accepts
+ */
+#define HAL_SW_DPPI_TASK_DIS_FROM_IDX(index) \
+	(NRF_DPPI_TASK_CHG0_DIS + index*(NRF_DPPI_TASK_CHG1_EN - NRF_DPPI_TASK_CHG0_EN))
```
Comment (cvinayak) [inline subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/radio/radio_nrf5_dppi.h:332]:
```suggestion
	(NRF_DPPI_TASK_CHG0_DIS + ((index) * (NRF_DPPI_TASK_CHG1_EN - NRF_DPPI_TASK_CHG0_EN)))
```

**Review by cvinayak — CHANGES_REQUESTED** [review]

File: subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/radio/radio_nrf5_dppi.h (hunk 1 above)
Comment (aescolar) [reply subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/radio/radio_nrf5_dppi.h]: done :)

**Review by cvinayak — APPROVED** [review]

**Review by kruithofa — APPROVED** [review]

**Comment by cvinayak** [comment]
Could you change commit title to "Bluetooth: Controller: ..."

**Comment by aescolar** [comment]
@kruithofa @cvinayak this last push is just the commit title change as requested by Vinayak. No changes otherwise.

**Review by cvinayak — APPROVED** [review]

**Pull request merged**